src/test/libcephfs/deleg.cc

```diff
@@ -166,7 +166,12 @@ static void simple_deleg_test(struct ceph_mount_info *cmount, struct ceph_mount_
   sprintf(filename, "deleg.rwrw.%x", getpid());
   ASSERT_EQ(ceph_ll_create(cmount, root, filename, 0666,
 		    O_RDWR|O_CREAT|O_EXCL, &file, &fh, &stx, 0, 0, perms), 0);
-  ASSERT_EQ(ceph_ll_delegation(cmount, fh, CEPH_DELEGATION_WR, dummy_deleg_cb, &recalled), 0);
+  int ret, retry = 0;
+  do {
+    ret = ceph_ll_delegation(cmount, fh, CEPH_DELEGATION_WR, dummy_deleg_cb, &recalled);
+    usleep(10000);
+  } while (ret == -EAGAIN && retry++ < 1000); /* Wait 10s at most */
+  ASSERT_EQ(ret, 0);
```

Comment:
It would be nice to have a helper function for this.

Reply:
Will fix it.